utils/frame/remote-externalities/src/lib.rs

```diff
@@ -230,33 +266,99 @@ impl<B: BlockT> Builder<B> {
 }
 
 // RPC methods
-impl<B: BlockT> Builder<B> {
+impl<B: BlockT> Builder<'_, B> {
 	async fn rpc_get_head(&self) -> Result<B::Hash, &'static str> {
 		trace!(target: LOG_TARGET, "rpc: finalized_head");
-		RpcApi::<B>::finalized_head(&self.as_online().rpc()).await.map_err(|e| {
+		RpcApi::<B>::finalized_head(self.as_online().rpc_client()).await.map_err(|e| {
 			error!("Error = {:?}", e);
 			"rpc finalized_head failed."
 		})
 	}
 
-	/// Relay the request to `state_getPairs` rpc endpoint.
+	/// Get all the keys at `prefix` at `hash` using the paged, safe RPC methods.
+	async fn get_keys_paged(
+		&self,
+		prefix: StorageKey,
+		hash: B::Hash,
+	) -> Result<Vec<StorageKey>, &'static str> {
+		const PAGE: u32 = 512;
```

Comment:
Can you explain this "max"  `PAGE` size?

AFAIU, the `state rpc` is using 1000 why is 512 used here?

Reply:
page size can be any value less than 1000, I randomly chose the biggest power of 2 in that range :D no reason, can also be 1000, can also be configurable, but neither is probably a big deal at this point.